### PR TITLE
Add rt1180 qtmr support

### PR DESCRIPTION
**Pull request by lucien-nxp**
add qtmer1_timer0 supported in counter_basic_api case for RT1180.

### CODE REVIEW

**Comment by lucien-nxp** [comment]
@hakehuang , could you help test counter_basic_api case on RT1180?

File: tests/drivers/counter/counter_basic_api/boards/mimxrt1180_evk_mimxrt1189_cm33.conf [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#
+# Copyright 2024 NXP
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+CONFIG_COUNTER_MCUX_QTMR=y
+CONFIG_USERSPACE=n
```
Comment (danieldegrasse) [inline tests/drivers/counter/counter_basic_api/boards/mimxrt1180_evk_mimxrt1189_cm33.conf:7]: Why disable userspace? Are we running out of MPU regions?

**Comment by hakehuang** [comment]
> @hakehuang , could you help test counter_basic_api case on RT1180?

board testing pass

```
2024-09-04 09:36:59,455 - twister - DEBUG - DEVICE: �*** Booting Zephyr OS build v3.7.0-1989-g57af539bbbf3 ***
2024-09-04 09:36:59,458 - twister - DEBUG - DEVICE: Running TESTSUITE counter_basic
2024-09-04 09:36:59,464 - twister - DEBUG - DEVICE: ===================================================================
2024-09-04 09:36:59,766 - twister - DEBUG - DEVICE: START - test_all_channels
2024-09-04 09:36:59,768 - twister - DEBUG - DEVICE: Testing gpt@2ec0000
2024-09-04 09:36:59,900 - twister - DEBUG - DEVICE: Testing timer0
2024-09-04 09:37:00,034 - twister - DEBUG - DEVICE: PASS - test_all_channels in 0.265 seconds
2024-09-04 09:37:00,040 - twister - DEBUG - DEVICE: ===================================================================
2024-09-04 09:37:00,044 - twister - DEBUG - DEVICE: START - test_cancelled_alarm_does_not_expire
2024-09-04 09:37:00,046 - twister - DEBUG - DEVICE: Skipped for gpt@2ec0000
2024-09-04 09:37:00,148 - twister - DEBUG - DEVICE: Skipped for timer0
2024-09-04 09:37:00,253 - twister - DEBUG - DEVICE: SKIP - test_cancelled_alarm_does_not_expire in 0.205 seconds
2024-09-04 09:37:00,259 - twister - DEBUG - DEVICE: ===================================================================
2024-09-04 09:37:00,261 - twister - DEBUG - DEVICE: START - test_late_alarm
2024-09-04 09:37:00,263 - twister - DEBUG - DEVICE: Skipped for gpt@2ec0000
2024-09-04 09:37:00,365 - twister - DEBUG - DEVICE: Skipped for timer0
2024-09-04 09:37:00,469 - twister - DEBUG - DEVICE: SKIP - test_late_alarm in 0.204 seconds
2024-09-04 09:37:00,475 - twister - DEBUG - DEVICE: ===================================================================
2024-09-04 09:37:00,477 - twister - DEBUG - DEVICE: START - test_late_alarm_error
2024-09-04 09:37:00,479 - twister - DEBUG - DEVICE: Skipped for gpt@2ec0000
2024-09-04 09:37:00,581 - twister - DEBUG - DEVICE: Skipped for timer0
2024-09-04 09:37:00,665 - twister - DEBUG - -- west flash: using runner linkserver
-- runners.linkserver: LinkServer: /usr/local/LinkServer/LinkServer, version v1.6.133

2024-09-04 09:37:00,685 - twister - DEBUG - DEVICE: SKIP - test_late_alarm_error in 0.204 seconds
2024-09-04 09:37:00,691 - twister - DEBUG - DEVICE: ===================================================================
2024-09-04 09:37:00,694 - twister - DEBUG - DEVICE: START - test_multiple_alarms
2024-09-04 09:37:00,696 - twister - DEBUG - DEVICE: Skipped for gpt@2ec0000
2024-09-04 09:37:00,798 - twister - DEBUG - DEVICE: Skipped for timer0
2024-09-04 09:37:00,902 - twister - DEBUG - DEVICE: SKIP - test_multiple_alarms in 0.204 seconds
2024-09-04 09:37:00,908 - twister - DEBUG - DEVICE: ===================================================================
2024-09-04 09:37:00,911 - twister - DEBUG - DEVICE: START - test_set_top_value_with_alarm
2024-09-04 09:37:00,914 - twister - DEBUG - DEVICE: E: Wrap can only be set to 0xffffffff
2024-09-04 09:37:00,917 - twister - DEBUG - DEVICE: Skipped for gpt@2ec0000
2024-09-04 09:37:01,020 - twister - DEBUG - DEVICE: E: Wrap can only be set to 0xffff
2024-09-04 09:37:01,021 - twister - DEBUG - DEVICE: Skipped for timer0
2024-09-04 09:37:01,126 - twister - DEBUG - DEVICE: SKIP - test_set_top_value_with_alarm in 0.211 seconds
2024-09-04 09:37:01,132 - twister - DEBUG - DEVICE: ===================================================================
2024-09-04 09:37:01,135 - twister - DEBUG - DEVICE: START - test_short_relative_alarm
2024-09-04 09:37:01,138 - twister - DEBUG - DEVICE: Skipped for gpt@2ec0000
2024-09-04 09:37:01,241 - twister - DEBUG - DEVICE: Testing timer0
2024-09-04 09:37:01,346 - twister - DEBUG - DEVICE: PASS - test_short_relative_alarm in 0.207 seconds
2024-09-04 09:37:01,352 - twister - DEBUG - DEVICE: ===================================================================
2024-09-04 09:37:01,355 - twister - DEBUG - DEVICE: START - test_single_shot_alarm_notop
2024-09-04 09:37:01,357 - twister - DEBUG - DEVICE: Testing gpt@2ec0000
2024-09-04 09:37:01,529 - twister - DEBUG - DEVICE: Testing timer0
2024-09-04 09:37:01,704 - twister - DEBUG - DEVICE: PASS - test_single_shot_alarm_notop in 0.344 seconds
2024-09-04 09:37:01,709 - twister - DEBUG - DEVICE: ===================================================================
2024-09-04 09:37:01,713 - twister - DEBUG - DEVICE: START - test_single_shot_alarm_top
2024-09-04 09:37:01,716 - twister - DEBUG - DEVICE: E: Wrap can only be set to 0xffffffff
2024-09-04 09:37:01,718 - twister - DEBUG - DEVICE: Skipped for gpt@2ec0000
2024-09-04 09:37:01,821 - twister - DEBUG - DEVICE: E: Wrap can only be set to 0xffff
2024-09-04 09:37:01,823 - twister - DEBUG - DEVICE: Skipped for timer0
2024-09-04 09:37:01,927 - twister - DEBUG - DEVICE: SKIP - test_single_shot_alarm_top in 0.211 seconds
2024-09-04 09:37:01,933 - twister - DEBUG - DEVICE: ===================================================================
2024-09-04 09:37:01,936 - twister - DEBUG - DEVICE: TESTSUITE counter_basic succeeded
2024-09-04 09:37:01,940 - twister - DEBUG - DEVICE: Running TESTSUITE counter_no_callback
2024-09-04 09:37:01,946 - twister - DEBUG - DEVICE: ===================================================================
2024-09-04 09:37:02,249 - twister - DEBUG - DEVICE: START - test_set_top_value_without_alarm
2024-09-04 09:37:02,253 - twister - DEBUG - DEVICE: E: Wrap can only be set to 0xffffffff
2024-09-04 09:37:02,255 - twister - DEBUG - DEVICE: Skipped for gpt@2ec0000
2024-09-04 09:37:02,358 - twister - DEBUG - DEVICE: E: Wrap can only be set to 0xffff
2024-09-04 09:37:02,360 - twister - DEBUG - DEVICE: Skipped for timer0
2024-09-04 09:37:02,465 - twister - DEBUG - DEVICE: SKIP - test_set_top_value_without_alarm in 0.211 seconds
2024-09-04 09:37:02,471 - twister - DEBUG - DEVICE: ===================================================================
2024-09-04 09:37:02,474 - twister - DEBUG - DEVICE: TESTSUITE counter_no_callback succeeded
2024-09-04 09:37:02,478 - twister - DEBUG - DEVICE: ------ TESTSUITE SUMMARY START ------
2024-09-04 09:37:02,487 - twister - DEBUG - DEVICE: SUITE PASS - 100.00% [counter_basic]: pass = 3, fail = 0, skip = 6, total = 9 duration = 2.055 seconds
2024-09-04 09:37:02,493 - twister - DEBUG - DEVICE: - PASS - [counter_basic.test_all_channels] duration = 0.265 seconds
2024-09-04 09:37:02,501 - twister - DEBUG - DEVICE: - SKIP - [counter_basic.test_cancelled_alarm_does_not_expire] duration = 0.205 seconds
2024-09-04 09:37:02,506 - twister - DEBUG - DEVICE: - SKIP - [counter_basic.test_late_alarm] duration = 0.204 seconds
2024-09-04 09:37:02,513 - twister - DEBUG - DEVICE: - SKIP - [counter_basic.test_late_alarm_error] duration = 0.204 seconds
2024-09-04 09:37:02,519 - twister - DEBUG - DEVICE: - SKIP - [counter_basic.test_multiple_alarms] duration = 0.204 seconds
2024-09-04 09:37:02,526 - twister - DEBUG - DEVICE: - SKIP - [counter_basic.test_set_top_value_with_alarm] duration = 0.211 seconds
2024-09-04 09:37:02,533 - twister - DEBUG - DEVICE: - PASS - [counter_basic.test_short_relative_alarm] duration = 0.207 seconds
2024-09-04 09:37:02,540 - twister - DEBUG - DEVICE: - PASS - [counter_basic.test_single_shot_alarm_notop] duration = 0.344 seconds
2024-09-04 09:37:02,547 - twister - DEBUG - DEVICE: - SKIP - [counter_basic.test_single_shot_alarm_top] duration = 0.211 seconds
2024-09-04 09:37:02,557 - twister - DEBUG - DEVICE: SUITE SKIP -   0.00% [counter_no_callback]: pass = 0, fail = 0, skip = 1, total = 1 duration = 0.211 seconds
2024-09-04 09:37:02,564 - twister - DEBUG - DEVICE: - SKIP - [counter_no_callback.test_set_top_value_without_alarm] duration = 0.211 seconds
2024-09-04 09:37:02,568 - twister - DEBUG - DEVICE: ------ TESTSUITE SUMMARY END ------
2024-09-04 09:37:02,574 - twister - DEBUG - DEVICE: ===================================================================
2024-09-04 09:37:02,577 - twister - DEBUG - DEVICE: RunID: d96573430b5cc710d7ceae2311e0364b
2024-09-04 09:37:02,580 - twister - DEBUG - DEVICE: PROJECT EXECUTION SUCCESSFUL
```

**Review by danieldegrasse — DISMISSED** [review]

**Review by decsny — DISMISSED** [review]

**Comment by dleach02** [comment]
@nordic-krch can you review this PR so we can get it merged?

**Review by dleach02 — DISMISSED** [review]

**Review by nordic-krch — DISMISSED** [review]

**Review by danieldegrasse — COMMENTED** [review]
We should add an entry to the board's supported table, right?
https://github.com/zephyrproject-rtos/zephyr/blob/8fda052826de630fc8a973325675759234a01fef/boards/nxp/mimxrt1180_evk/doc/index.rst#L94

**Comment by lucien-nxp** [comment]
> We should add an entry to the board's supported table, right?
> 
> https://github.com/zephyrproject-rtos/zephyr/blob/8fda052826de630fc8a973325675759234a01fef/boards/nxp/mimxrt1180_evk/doc/index.rst#L94

Updated in doc/index.rst

**Review by danieldegrasse — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Review by nordic-krch — APPROVED** [review]

**Pull request merged**